File: Java/Easy/WrongAnswer/minimum-operations-to-make-the-array-increasing_6.java
class Solution {
    public int minOperations(int[] nums) {
        
        int n = nums.length;
        int[] dp = new int[n];
        dp[0] = 0;
        for (int i = 1; i < n; i++) {
            dp[i] = Math.max(dp[i - 1] + 1 - nums[i], 0) + dp[i - 1];
            nums[i] = Math.max(nums[i - 1] + 1, nums[i]);
        }
        return dp[n - 1];
    }
}

File: Rust/Easy/Accepted/minimum-operations-to-make-the-array-increasing_6.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        
        let mut ret = 0;
        let mut pre = nums[0];
        for i in 1..nums.len() {
            if nums[i] <= pre {
                ret += pre - nums[i] + 1;
                pre += 1;
            } else {
                pre = nums[i];
            }
        }
        ret
    }
}